sdk/core/System.ClientModel/src/Internal/Logging/ClientModelEventSource.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Tracing;
+using System.Text;
+
+namespace System.ClientModel.Internal;
+
+// The methods in this class should only ever be called from LoggingHandler
```

Comment:
Usually, this file would include checks to see if the given level is enabled before writing the event. However, those checks are now in the LoggingHandler class and we don't want to do the checks multiple times.

Just noting that this is not a typical pattern.